plasmapy/dispersion/numerical/stix_.py

```diff
@@ -0,0 +1,269 @@
+"""
+This module contains functionality for calculating the numerical
+solutions to the Stix cold plasma function.
+"""
+__all__ = ["stix"]
+
+import astropy.units as u
+import numpy as np
+
+from astropy.constants.si import c
+
+from plasmapy.formulary.frequencies import gyrofrequency, plasma_frequency
+from plasmapy.particles import Particle, ParticleList
+from plasmapy.utils.decorators import validate_quantities
+
+c_si_unitless = c.value
+
+
+@validate_quantities(
+    B={"can_be_negative": False},
+    n_i={"can_be_negative": False},
+    w={"can_be_negative": False, "can_be_zero": False},
+)
+def stix(
+    B: u.T,
+    w: u.rad / u.s,
+    ions: Particle,
+    n_i: u.m ** -3,
+    theta: u.rad,
+):
+    r"""
+    Calculate the cold plasma function solution by using
+    :cite:t:`bellan:2012`, this uses the numerical method to find the
+    wave number(s), (:math:`k`), for the dispersion relation provided by
+    :cite:t:`stringer:1963`. This dispersion relation also assumes
+    uniform magnetic field :math:`\mathbf{B_0}`, :math:`\theta` is the
+    angle between the magnetic and the normal surface of the wave
+    vector. For more information see the **Notes** section below.
```

Comment:
```suggestion
    Calculate the cold plasma dispersion function presented by
    :cite:t:`stix:1992`, and discussed by :cite:t:`bellan:2012`.
    This is an analytical solution of equation 8 in
    :cite:t:`bellan:2012`.  See the **Notes** section below for
    additional details.
```

It dawned on me that solving for `k` is now an analytical solution, so we should move this functionality to `plasmapy.dispersion.analytical`.